### ISSUE TITLE
[Enhancement]: Make "Used for variations" checkbox visible for other variable-ish product types.

### ISSUE THREAD

**Issue by helgatheviking** [woothemes/woocommerce#35004]
### Describe the solution you'd like

When creating a custom product type that extends the variable product type, Need a way to show the `show_if_variable` attribute checkbox . The classes cannot be altered in the [`html-product-attribute.php` template](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js#L716):

https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/views/html-product-attribute.php#L85-L87

But there's not a specific trigger for when the [save attributes](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js#L742) ajax function completes. 

### Describe alternatives you've considered

Looks like you can listen for `$( '#variable_product_options' ).on( 'reload', function() {} );` but that's not initially obvious.

**Comment by barryhughes** [comment]
Hi @helgatheviking,

> But there's not a specific trigger for when the [save attributes](https://github.com/woocommerce/woocommerce/issues/url) ajax function completes.

I think you were referencing a piece of code, but the link seems to be broken. Can you confirm, though, that existing event [`woocommerce_added_attribute`](https://github.com/woocommerce/woocommerce/blob/7.0.0/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js#L468) isn't suitable for your needs here—what would you prefer to see if so?

**Comment by helgatheviking** [comment]
@barryhughes Whoops... I think that link is in the metaboxes scripts, when the attributes are [reloaded from the ajax response](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js#L742).

Unless it's possible to use this [`reload()`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js#L798)

**Comment by barryhughes** [comment]
Gotcha—thanks for clarifying, that makes perfect sense. I'm sure we can add a trigger at the bottom of that callback (and if you wish to submit a PR feel free to request my review—otherwise we will work our way towards this).

**Comment by helgatheviking** [comment]
@barryhughes there it is: https://github.com/woocommerce/woocommerce/pull/36943 Though I will say I _Was_ able to do what I needed on
```
    $( '#variable_product_options' ).on(
        'reload',
        function() {
            console.debug("attributes saved");
        } 
    );
    ```
    
So I'm no longer blocked by this and it would be ok to close it as `wontfix`. i don't think it hurts anything though and 1 line of a new trigger won't have any backcompat issues.

**Comment by barryhughes** [comment]
Agree, there is still utility and value in a dedicated event.
